src/interface/blas1_interface.hpp

```diff
@@ -258,17 +360,14 @@ template <typename sb_handle_t, typename container_t, typename ContainerI,
 typename sb_handle_t::event_t _iamax(
     sb_handle_t &sb_handle, index_t _N, container_t _vx, increment_t _incx,
     ContainerI _rs, const typename sb_handle_t::event_t &_dependencies) {
-  typename VectorViewType<container_t, index_t, increment_t>::type vx =
-      make_vector_view(_vx, _incx, _N);
-  auto rs = make_vector_view(_rs, static_cast<increment_t>(1),
-                             static_cast<index_t>(1));
-  const auto localSize = sb_handle.get_work_group_size();
-  const auto nWG = 2 * localSize;
-  auto tupOp = make_tuple_op(vx);
-  auto assignOp = make_assign_reduction<IMaxOperator>(rs, tupOp, localSize,
-                                                      localSize * nWG);
-  auto ret = sb_handle.execute(assignOp, _dependencies);
-  return ret;
+  if (_incx < 0) {
```

Comment:
Thank you for you work @muhammad-tanvir-1211 
oneMKL specification says: *"If either n or incx is not positive, the routine returns 0, regardless of the base of the index selected."* 
So I think we should add a check also for `_N` here and at line `385`